src/ksm/group.go

```diff
@@ -4,52 +4,52 @@ import (
 	"fmt"
 
 	"github.com/newrelic/infra-integrations-sdk/log"
+	corev1 "k8s.io/api/core/v1"
 
-	"github.com/newrelic/nri-kubernetes/v2/src/client"
 	"github.com/newrelic/nri-kubernetes/v2/src/data"
 	"github.com/newrelic/nri-kubernetes/v2/src/definition"
-	"github.com/newrelic/nri-kubernetes/v2/src/ksm/metric"
 	"github.com/newrelic/nri-kubernetes/v2/src/prometheus"
 )
 
 type ksmGrouper struct {
-	queries   []prometheus.Query
-	client    client.HTTPGetter
-	logger    log.Logger
-	k8sClient client.Kubernetes
+	queries                []prometheus.Query
+	filteredMetricFamilies prometheus.FilteredMetricFamilies
+	logger                 log.Logger
+	services               []*corev1.Service
```

Comment:
Perhaps my IDE is on drugs, but I'm not being able to find any place where this field is _written_ to. Is this intended?

Reply:
this is a bug

Reply:
Fixed. Morevoer grouper now receives the lister directly.